### PR TITLE
Fix cart and product page layout issues.

### PR DESCRIPTION
**Pull request by itsDiwaker**
- Update card html to correct one.
- Re-arrange layout structure.

### CODE REVIEW

**Comment by itsDiwaker** [comment]
@CristinaSolana 
Below fixes address issues reported by @cristianPerez on pull request #229. They are not browser specific issues so created a separate branch and pull request.

**For product page:**

- Interesting items section html has been updated.
- Product page media section and info section has been re-arranged using grid-container instead of margins and offsets.

**For Cart page:**

- Interesting items section html has been updated.

**Comment by itsDiwaker** [comment]
@CristinaSolana, 

this is the pull request that addresses safari issues reported by Christian on #229

Also, I have resolved conflict with `product.liquid`. Please note that `product.liquid` from this pull request contains major layout fixes. 
So we need to check whether all the product page functionalities are intact before merging.

Let me know if you find any issue related to layout or functionality that can result from this merging/refactoring.

File: templates/product.liquid [hunk 1]
```diff
@@ -5,352 +5,515 @@
...
-                <div class="grid-x">
-                    {% if websiteSettings.storefrontPayments.uom != 'none' %}
-                    <div class="cell small-6 medium-12 medium-order-2 product-detail-points">
-                        {% if websiteSettings.storefrontPayments.uom == 'points' %}
-                        {{item.priceMin | ceil | times: websiteSettings.storefrontPayments.pointConversion}} {% if item.priceMin != item.priceMax %}- {{item.priceMax | ceil | times: websiteSettings.storefrontPayments.pointConversion}}{% endif %} Points
-                        {% else if websiteSettings.storefrontPayments.uom == 'dollars' %}
-                        {{item.priceMin | money: 'USD'}} {% if item.priceMin != item.priceMax %}- {{item.priceMax | money: 'USD'}}{% endif %}
+            <div class="product-detail-mobile-gallery"></div>
+
+            <style>
```
Comment (itsDiwaker) [inline templates/product.liquid]: @CristinaSolana  
this inline style came from upstream.

Comment (CristinaSolana) [reply templates/product.liquid]: @itsDiwaker Np, we can improve all that later. :)

**Pull request merged**